game/src/ecs/components/ai/fight/RangeAI.java

```diff
@@ -0,0 +1,89 @@
+package ecs.components.ai.fight;
+
+import com.badlogic.gdx.ai.pfa.GraphPath;
+import ecs.components.MissingComponentException;
+import ecs.components.PositionComponent;
+import ecs.components.ai.AITools;
+import ecs.components.skill.Skill;
+import ecs.entities.Entity;
+import level.elements.tile.Tile;
+import starter.Game;
+import tools.Point;
+
+import java.util.List;
+
+import static com.badlogic.gdx.math.MathUtils.random;
+import static ecs.components.ai.AITools.getAccessibleTilesInRange;
+
+
+public class RangeAI implements IFightAI{
+
+    private final float attackRange;
+    private final float distance;
+    private final Skill skill;
+    boolean isNotBetweenAttackAndDistance = true;
+    private GraphPath<Tile> path;
+
+    /**
+     * Attacks the player if he is within the given range between attackRange and distance. Otherwise, it will move into that range.
+     *
+     * @param attackRange max. Range in which the attack skill should be executed
+     * @param distance min. Range in which the attack skill should be executed
+     * @param skill Skill to be used when an attack is performed
+     */
+
+    public RangeAI(float attackRange, float distance, Skill skill){
+        this.attackRange = attackRange;
+        this.distance = distance;
+        this.skill = skill;
+    }
+
+    @Override
+    public void fight(Entity entity) {
+        if(attackRange > distance) {
+            if (AITools.playerInRange(entity, distance) && isNotBetweenAttackAndDistance) {
+                isNotBetweenAttackAndDistance = false;
+                Point positionHero = getPosition(Game.getHero().get());
+                Point positionEntity = getPosition(entity);
+                List<Tile> tiles = getAccessibleTilesInRange(positionEntity, attackRange - distance);
+                Point newPosition;
+                boolean newPositionFound = false;
+                for (int index = 0; index < tiles.size(); index++) {
+                    newPosition = tiles.get(index).getCoordinate().toPoint();
+                    if (!AITools.inRange(newPosition, positionHero, distance)) {
+                        path = AITools.calculatePath(positionEntity, newPosition);
+                        newPositionFound = true;
+                        break;
+                    }
+                }
```

Comment:
Wäre die Schleife nicht besser, wenn sie das Tile sucht, welches am weitesten von Hero weg ist, aber immer noch in der Reichweite zum Angreifen? (Nur so als Gedanke)

Reply:
Würde man von Helden aus einfach einen Abstand ermitteln, dann könnte es sein, dass das Monster durch den Helden durch läuft.